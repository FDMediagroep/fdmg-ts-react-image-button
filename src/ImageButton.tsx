import * as React from 'react';
import {MouseEvent} from "react";

export interface Props {
    className?: string;
    onClick?: (event: MouseEvent<HTMLImageElement>) => void;
    alt: string;
    src: string;
}

/**
 * Renders an image with an onClick hook for parent component.
 */
export default class ImageButton extends React.Component<Props, any> {
    public state: any;
    public props: Props;

    constructor(props: Props) {
        super(props);
        this.props = props;
        console.info('ImageButton component', props);
    }

    componentDidMount() {
        console.info('ImageButton mounted');
    }
        
    componentWillUnmount() {
        console.info('ImageButton will unmount');
    }

    componentWillReceiveProps(nextProps) {
        console.info('ImageButton next props', nextProps);
    }

    render() {
        return (
            <img 
                onClick={this.props.onClick}
                src={this.props.src}
                alt={this.props.alt}
                className={this.props.className}
                role={this.props.onClick?'button':undefined}
            />
        );
    }
}
