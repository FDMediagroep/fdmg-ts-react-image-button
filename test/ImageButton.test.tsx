import * as Enzyme from 'enzyme';
import * as Adapter from 'enzyme-adapter-react-16';
import { shallow, mount } from 'enzyme';

import * as React from 'react';
import * as renderer from 'react-test-renderer';
import ImageButton from "../src/ImageButton";

console.info = function() {};

beforeAll(() => {
    Enzyme.configure({ adapter: new Adapter() });
});

test('ImageButton renders correctly', () => {
    let component = renderer.create(
        <ImageButton onClick={()=>{}} alt={"alt-text"} src={'test.png'} className={'css-class-name'}/>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    component = renderer.create(
        <ImageButton alt={"alt-text"} src={'test.png'} className={'css-class-name'}/>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

test('ImageButton buttons clicked', () => {
    const spy = jest.fn();
    let component = mount(
        <ImageButton onClick={spy} alt={"alt-text"} src={'test.png'} className={'css-class-name'}/>
    );

    component.simulate('click');
    expect(spy).toHaveBeenCalled();
});
