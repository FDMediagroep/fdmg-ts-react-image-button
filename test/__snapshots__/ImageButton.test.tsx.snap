// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageButton renders correctly 1`] = `
<img
  alt="alt-text"
  className="css-class-name"
  onClick={[Function]}
  role="button"
  src="test.png"
/>
`;

exports[`ImageButton renders correctly 2`] = `
<img
  alt="alt-text"
  className="css-class-name"
  onClick={undefined}
  role={undefined}
  src="test.png"
/>
`;
